lib/prometheus_exporter/client.rb

```diff
@@ -234,8 +241,7 @@ def ensure_socket!
 
       nil
     rescue StandardError
-      @socket = nil
```

Comment:
Previously, we were just assigning the `@socket` to `nil` which causes a `TCPSocket` to remain open.

Reply:
No the worker loop is meant to ensure that a socket is opened and reuse the socket across loops. Is there a reason you think we should close the socket each time we process a message?

Reply:
Good catch and I've made the change to call `close_socket!`. However, I don't think we can easily test for this and I think it is OK since the risk of dropping a message is low.